scope_registry.go

```diff
@@ -86,19 +86,27 @@ func (r *scopeRegistry) Subscope(parent *scope, prefix string, tags map[string]s
 		return NoopScope.(*scope)
 	}
 
-	key := scopeRegistryKey(prefix, parent.tags, tags)
-
+	preSanitizeKey := keyForPrefixedStringMapsAsKey(prefix, parent.tags, tags)
 	r.mu.RLock()
-	if s, ok := r.lockedLookup(key); ok {
+	if s, ok := r.lockedLookup(preSanitizeKey.CastAsString()); ok {
 		r.mu.RUnlock()
+		preSanitizeKey.Release()
 		return s
 	}
 	r.mu.RUnlock()
 
+	preSanitizeKeyStr := preSanitizeKey.CopyAsString()
+	preSanitizeKey.Release()
```

Comment:
If you decide to leave the `bufferPool` in place, I would encourage you to add a test that verifies the behavior under high concurrency. It increases our confidence that the objects are released correctly, even when we later want to change the code (e.g., to leverage the Go generics)

Reply:
yes, this was the reason I kept it as object pool instead of trying your suggestion. If I have to make byte slice on the stack, the method has to be inlined for it to work. However the same method is used from other places as well and it would need duplication of logic.